debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresErrorHandler.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.postgresql;
+
+import java.net.SocketException;
+
+import io.debezium.connector.base.ChangeEventQueue;
+import io.debezium.pipeline.ErrorHandler;
+import io.debezium.util.Throwables;
+
+/**
+ * Error handler for Postgres.
+ *
+ * @author Gunnar Morling
+ */
+public class PostgresErrorHandler extends ErrorHandler {
+
+    public PostgresErrorHandler(String logicalName, ChangeEventQueue<?> queue, Runnable onThrowable) {
+        super(PostgresConnector.class, logicalName, queue, onThrowable);
+    }
+
+    @Override
+    protected boolean isRetriable(Throwable throwable) {
```

Comment:
Wouldn't it be better to match against the Throwables that the JDBC driver exposes? I'm assuming that connection failures can happen with different root causes?

Reply:
I'm not quite sure I'm following. This connector-specific handler receives the raised exception and can act on it as required, by examining the root exception or the one raised. Could you elaborate what you have in mind?

Reply:
@gunnarmorling I think @hashhar want sot restart on any `SqlException`

Reply:
Ah, I see. I'm not so sure, I'd rather start with a specific set, we always can expand. Let's say there's an authorization failure (wrong password etc.), restarting doesn't make sense.

Reply:
Any SqlException is too extreme. But I'm thinking something like `org.postgresql.util.PSQLException: Database connection failed when writing to copy`. If PostgreSQL exposes specific exceptions for specific errors then that'll be useful. Not sure if the JDBC driver exposes those.

Reply:
But I agree with your point too. Start with specific ones and if in future a common parent is found then adjust accordingly.

Reply:
Looking for the `PSQLException` with that message as suggested now.